src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImpl.java

```diff
@@ -0,0 +1,182 @@
+                                                                   externalObjectDirectory.getId(),
+                                                                   armResponseUploadFileRecord.getMd5(), objectChecksum);
+            updateOsrIngestStatusToFailure(osr, checksumValidationFailedMessage);
+        });
+    }
+
+    @Override
+    protected void processUploadFileDataFailure(ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                                UploadFileFilenameProcessor uploadFileFilenameProcessor,
+                                                ExternalObjectDirectoryEntity externalObjectDirectory) {
+        super.processUploadFileDataFailure(armResponseUploadFileRecord, uploadFileFilenameProcessor, externalObjectDirectory);
+        getObjectStateRecord(externalObjectDirectory.getId())
+            .ifPresent(osr -> updateOsrIngestStatusToFailure(osr, armResponseUploadFileRecord.getExceptionDescription()));
+    }
+
+    @Override
+    protected void processInvalidLineFileActions(ArmResponseInvalidLineRecord armResponseInvalidLineRecord,
+                                                 ExternalObjectDirectoryEntity externalObjectDirectory) {
+        //tested
+        super.processInvalidLineFileActions(armResponseInvalidLineRecord, externalObjectDirectory);
+        getObjectStateRecord(externalObjectDirectory.getId())
+            .ifPresent(osr -> updateOsrIngestStatusToFailure(osr, armResponseInvalidLineRecord.getExceptionDescription()));
+    }
+
+
+    private Optional<ObjectStateRecordEntity> getObjectStateRecord(int armEod) {
+        Optional<ObjectStateRecordEntity> osrOptional = osrRepository.findByArmEodId(String.valueOf(armEod));
+        if (osrOptional.isEmpty()) {
+            log.error("Object State Record not found for Arm EOD {}", armEod);
+        }
+        return osrOptional;
+    }
+
+    private void updateOsrResponseReceivedAttributes(ObjectStateRecordEntity osr) {
+        osr.setFlagRspnRecvdFromArml(true);
+        osr.setDateRspnRecvdFromArml(timeHelper.currentOffsetDateTime());
+        osrRepository.save(osr);
+    }
+
+    private void deleteBlobDataAndEod(ExternalObjectDirectoryEntity armEod, ObjectStateRecordEntity osr) {
+        List<ExternalObjectDirectoryEntity> detsEods = externalObjectDirectoryRepository.findExternalObjectDirectoryByLocation(
+            EodHelper.detsLocation(),
+            armEod.getMedia(),
+            armEod.getTranscriptionDocumentEntity(),
+            armEod.getAnnotationDocumentEntity(),
+            armEod.getCaseDocument());
+
+        if (detsEods.size() == 1) {
+            boolean deleted = detsDataStoreDeleter.delete(detsEods.getFirst());
+            if (deleted) {
+                osr.setFlagFileDetsCleanupStatus(true);
+                osr.setDateFileDetsCleanup(timeHelper.currentOffsetDateTime());
+                osrRepository.save(osr);
```

Comment:
@hemantasharma1129  If the DETS data store has been deleted should the EOD record also be deleted or should it remain as the OSR record has a reference to the EOD record?

Reply:
yes hard delete on DETS Blobstorage and EOD table.